app/validators/email_validator.rb

```diff
@@ -1,9 +1,16 @@
+require 'mail'
 class EmailValidator < ActiveModel::EachValidator
-  EMAIL_REGEXP = /\A(([\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~]+((\.\"[\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+(\.[\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+)*\")*\.[\w!#\$%&\'\*\+\-\/=\?\^`\{\|\}~]+)*)|(\"[\w !#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+(\.[\w !#\$%&\'\*\+\-\/=\?\^`\{\|\}~\"\(\),:;<>@\[\\\] ]+)*\"))
-             @[a-z0-9]+((\.[a-z0-9]+)*|(\-[a-z0-9]+)*)*\z/ix
-  def validate_each(record, attribute, value)
+  def validate_each(record,attribute,value)
     return if options[:allow_blank] && value.empty?
     record.errors.add(attribute, _("is too long (maximum is 254 characters)")) if value && value.length > 254
-    record.errors.add(attribute, _("is invalid")) unless value && value.match(EMAIL_REGEXP)
+    begin
+      address = value.split("@")
+      encoded_address = address.count == 2 ? Mail::Encodings.decode_encode(address[0],:encode) + '@' + Mail::Encodings.decode_encode(address[1],:encode) : value
```

Comment:
@domcleal When you pass non-ascii address like  `Pelé@a.com`, the return value discards the `@`.
encoding each part (local and domain) separately, seems to work in gmail configuration, but encode the whole address at once doesn't. 

You can see in the mail gem [test](https://github.com/mikel/mail/pull/910/files#diff-0ca45d418776bf09983e700d422b262fR596), the address is split with `@` for supporting utf-8 encoding